### PR TITLE
iOS Device:  Add runtime model

### PR DESCRIPTION
**Pull request by jmoody**
## TL;DR

Some information about the device and app under test can only be extracted at runtime via a server query.

#### iOS Simulators

Many of the Calabash::Cucumber::Device class methods can be replaced be by inspecting the RunLoop::Device class.

#### Physical Devices

Most of the Calabash::Cucumber::Device class methods could be replaced _if_ we could use a third-party tool like `libimobiledevice`.

## Discussion

Some information about the device and app under test can only be extracted at runtime via a server query.

It is not possible to answer all of these questions for physical devices because the UDID of the device is opaque.  We can't know, for example, if the device is an iPad or iPhone 6.

#### Property: screen dimensions

This is only necessary for iOS < 8.0.  It is otherwise unused.  It can only be determined at runtime because of iPhone 6 and iPhone 6 plus scaling and sampling factors.

####  Property: iPhone apps emulated on iPads

On simulators, we could inspect the Info.plist and the RunLoop::Device to determine if this is an iPhone app emulated on an iPad.

On devices, we would need to used something like `ideviceinfo`.

#### Property: The version of the calabash server.

On iOS Simulators, this can be discovered before launching the app.

On device, it will not be possible _unless_ we can inspect the app bundle installed on the device before we launch.    For the calabash.framework, we could ship with an Info.plist that has the server version.  For calabash dylibs, we would need to be able to call `strings` and `grep` to get the server version.  I don't think this is possible on a physical device.

#### Property: form factor

For iOS simulators, this can be determined by inspecting the RunLoop::Device.

On devices, we'd need to something like `ideviceinfo` to determine the form factor.

#### Property: iPod?

There are no iPod simulators, so this method only makes sense on devices.

See the discussion about device form factors above.

### CODE REVIEW

**Comment by jmoody** [comment]
CC: @krukow @kamstrup @rasmuskl

File: lib/calabash/ios/device.rb [hunk 1]
```diff
@@ -378,6 +380,38 @@ def uninstall_app_on_physical_device(application, device_udid)
 
       private
 
+      attr_reader :runtime_info
+
+      # @!visibility private
+      def method_missing(name, *args, &block)
```
Comment (jmoody) [inline lib/calabash/ios/device.rb]: @TobiasRoikjer 

This is the madness that I was talking about.  I am not sure yet what information will need to be extracted at runtime.  If you see the discussion section of this PR, you'll see an analysis of the info that _must_ be obtained at runtime.
